### PR TITLE
Add vcard generator and insert tag

### PR DESCRIPTION
**Pull request by Xirdion**
Hi there. With this PR I would like to add the functionality to dynamically generate a vcard from the company data.
Currently there is only a inserttag (`company::vcard_url`) to generate the url to the vcard controller.

### CODE REVIEW

File: src/Controller/vCardController.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     */
+    public function download(Request $request): Response
+    {
+        // Try to load the current contao page
+        $pageId = (int) $request->query->get('contaoPage');
```
Comment (zoglo) [inline src/Controller/vCardController.php]: The parameter in this route doesn't match the one within the inserttag ('redirect').
I'd suggest changing it to something like 'page' in both files.

File: src/EventListener/InsertTagsListener.php [hunk 2]
```diff
@@ -155,6 +163,16 @@ private function replaceCompanyInsertTags($insertTag, $field)
...
+            case 'vcard_url':
+                $pageId = 0;
+                $request = $this->requestStack->getCurrentRequest();
+                if (null !== $request) {
+                    /** @var PageModel $page */
+                    $page = $request->attributes->get('pageModel');
+                    $pageId = $page->id;
+                }
+
+                return $this->router->generate('contao_company_vcard_download', ['redirect' => $pageId]);
```
Comment (zoglo) [inline src/EventListener/InsertTagsListener.php]: Same as in vCardController:
- parameter should be changed to e.g. 'page'. It does not match right now, hence throwing an error

File: src/Generator/vCardGenerator.php [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        if (null === $uuid) {
+            return $this;
+        }
+
+        $file = FilesModel::findByUuid($uuid);
+        if (null === $file) {
+            return $this;
+        }
+
+        $this->vCard->addLogo($file->getAbsolutePath());
```
Comment (zoglo) [inline src/Generator/vCardGenerator.php]: I would change this line to:

`$projectDir = System::getContainer()->getParameter('kernel.project_dir');`
`$this->vCard->addLogo(Path::makeAbsolute($file->path, $projectDir));`

to make sure that the bundle still works with Contao 4.9

**Review by zoglo — APPROVED** [review]

**Comment by zoglo** [comment]
Thank you @Xirdion,

I did fix a few things and made sure it's backwards compatible with Contao 4.9 LTS.
Added in v.1.3.3: https://github.com/oveleon/contao-company-bundle/releases/tag/1.3.3

File: src/Controller/vCardController.php (hunk 1 above)
Comment (Xirdion) [reply src/Controller/vCardController.php]: Oh, I forgot to update the parameter after I changed it.

**Pull request merged**